cloudferry/actions/compute/vm_restore.py

```diff
@@ -34,19 +34,22 @@ def run(self, info, **kwargs):
 
 
 class VmRestore(action.Action):
-    def run(self, info_backup, **kwargs):
+    def run(self, **kwargs):
         src_compute = self.src_cloud.resources[utils.COMPUTE_RESOURCE]
         dst_compute = self.dst_cloud.resources[utils.COMPUTE_RESOURCE]
         for instance_id in set(dst_compute.processing_instances +
                                dst_compute.failed_instances):
             LOG.debug("Delete VM %s on DST", instance_id)
             dst_compute.force_delete_vm_by_id(instance_id)
-        for instance_id in src_compute.processing_instances:
-            instance = info_backup['instances'][instance_id]['instance']
-            LOG.debug("Status of '%s' (%s) is changed to original state %s on "
-                      "SRC", instance['name'], instance_id, instance['status'])
-            # reset status of vm
-            src_compute.change_status(
-                instance['status'],
-                instance_id=instance_id)
+        info_backup = kwargs.get('info_backup')
+        if info_backup:
+            for instance_id in src_compute.processing_instances:
+                instance = info_backup['instances'][instance_id]['instance']
+                LOG.debug("Status of '%s' (%s) is changed to original state "
+                          "%s on SRC", instance['name'], instance_id,
+                          instance['status'])
+                # reset status of vm
+                src_compute.change_status(
```

Comment:
This loop will be interrupted if change status will raise some exception. You should catch them, but don't interrupt the loop